packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts

```diff
@@ -36,114 +47,85 @@ describe('trackLoadingTime', () => {
-    const { getViewUpdate, getViewUpdateCount } = viewTest
-
-    expect(getViewUpdateCount()).toEqual(1)
+    loadType = ViewLoadingType.INITIAL_LOAD
+    const { domMutationObservable, clock } = setupBuilder.build()
 
     clock.tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
 
-    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [
-      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION, {
-        loadEventEnd: LOAD_EVENT_AFTER_ACTIVITY_TIMING,
-      }),
-    ])
-
+    setLoadEvent(LOAD_EVENT_AFTER_ACTIVITY_TIMING)
     domMutationObservable.notify()
     clock.tick(AFTER_PAGE_ACTIVITY_END_DELAY)
 
-    clock.tick(THROTTLE_VIEW_UPDATE_PERIOD)
-
-    expect(getViewUpdateCount()).toEqual(2)
-    expect(getViewUpdate(1).commonViewMetrics.loadingTime).toEqual(LOAD_EVENT_AFTER_ACTIVITY_TIMING)
+    expect(loadingTimeCallback).toHaveBeenCalledTimes(1)
+    expect(loadingTimeCallback).toHaveBeenCalledWith(LOAD_EVENT_AFTER_ACTIVITY_TIMING)
   })
 
   it('should use computed loading time for initial view when load event is smaller than computed loading time', () => {
-    const { lifeCycle, domMutationObservable, clock } = setupBuilder.build()
-    const { getViewUpdate, getViewUpdateCount } = viewTest
-
-    expect(getViewUpdateCount()).toEqual(1)
+    loadType = ViewLoadingType.INITIAL_LOAD
+    const { domMutationObservable, clock } = setupBuilder.build()
 
     clock.tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
-    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [
-      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION, {
-        loadEventEnd: LOAD_EVENT_BEFORE_ACTIVITY_TIMING,
-      }),
-    ])
+
+    setLoadEvent(LOAD_EVENT_BEFORE_ACTIVITY_TIMING)
+
     domMutationObservable.notify()
     clock.tick(AFTER_PAGE_ACTIVITY_END_DELAY)
-    clock.tick(THROTTLE_VIEW_UPDATE_PERIOD)
 
-    expect(getViewUpdateCount()).toEqual(2)
-    expect(getViewUpdate(1).commonViewMetrics.loadingTime).toEqual(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+    expect(loadingTimeCallback).toHaveBeenCalledTimes(1)
+    expect(loadingTimeCallback).toHaveBeenCalledWith(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
   })
 
   it('should use computed loading time from time origin for initial view', () => {
+    loadType = ViewLoadingType.INITIAL_LOAD
+    const { domMutationObservable, clock } = setupBuilder.build()
+
     // introduce a gap between time origin and tracking start
     // ensure that `load event > activity delay` and `load event < activity delay + clock gap`
     // to make the test fail if the clock gap is not correctly taken into account
     const CLOCK_GAP = (LOAD_EVENT_AFTER_ACTIVITY_TIMING - BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY + 1) as Duration
 
-    setupBuilder.clock!.tick(CLOCK_GAP)
-
-    const { lifeCycle, domMutationObservable, clock } = setupBuilder.build()
-    const { getViewUpdate, getViewUpdateCount } = viewTest
-
-    expect(getViewUpdateCount()).toEqual(1)
+    clock.tick(CLOCK_GAP)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏According to the comment above, the `clock.tick(CLOCK_GAP)` should be done before `setupBuilder.build()`. Maybe revert to the previous implem?

Reply:
Hum, I checked and even with the previous version of the test if you move `clock.tick(CLOCK_GAP)` after `setupBuilder.build()` it still works..

Reply:
Yes. The point of the test is to make sure the loading time is relative to `timeOrigin` and not relative to the time `trackLoadingTime` was started. The `clock.tick(CLOCK_GAP)` was there to emulate a `trackLoadingTime()` invoked a bit after `timeOrigin`.

Reply:
I fixed the test